pkg/cloud/awscloud/awscloud.go

```diff
@@ -388,6 +385,18 @@ func (a *AWS) Register(name, bucket, key string, shareWith []string, rpmArch str
 	return registerOutput.ImageId, snapshotID, nil
 }
 
+// Cleanup removes any temporary resources created during the image registration process.
+// Make sure to call this function via defer to ensure cleanup is always performed.
+// It will return an error if resources no longer exist or if there was an error deleting them.
+func (a *AWS) Cleanup(name, bucket, key string) error {
```

Comment:
Should we call this "DeleteObject" as this is what it is doing? Why is "name" passed in? AFAICT it is not used?

Reply:
Since I saw an interface in bib codebase, I got impression there is an idea to have a common uploader interface at some point. Azure also has a cleanup similar to S3, tho, in that case this is completely done by a caller.

Idiomatic Go is to create an interface called `Cleaner` with this only method and then clients can use that to perform cleaning. I dontt mind changing this to anything else, but there is `Closer` in the standard library as well as `Reader` as well. For `DeleteObject` we would need to call this `Deleter`. But what if in the future cleaning involves more steps not just deleting "objects"?